Client.py

```diff
@@ -0,0 +1,240 @@
+                if len(parsed_lines) == 1:
+                    self._recv_buff = ""
+                else:
+                    self._recv_buff = parsed_lines[1]
+
+            # If we do have request line, get headers
+            if self.request is not None:
+                if not self.request.full_request and \
+                        constants.DOUBLE_CRLF in self._recv_buff:
+                    self.request.full_request = True
+                self.logger.debug("start getting lines")
+                self._recv_lines()
+                self.logger.debug("after recv %s" % self._recv_buff)
+                self.logger.debug("got lines")
+                if constants.CRLF in self._recv_buff:
+                    self.logger.debug(self._recv_buff)
+                    lines = self._recv_buff.split(constants.DOUBLE_CRLF)[
+                        0].split(constants.CRLF)
+                    for line in lines[:-1]:
+                        if ": " in line:
+                            self._parse_header(line)
+                    if ": " in lines[-1]:
+                        self._parse_header(lines[-1])
+                        self._recv_buff = ""
+                    else:
+                        self._recv_buff = lines[-1]
+
+                    self.logger.debug("Now recv_buff is %s" % self._recv_buff)
+
+        except CustomExceptions.Disconnect:
+            raise
+        except OSError as e:
+            self.logger.error(traceback.format_exc())
+            if e.errno == errno.ENOENT:
+                self._send_buff = util.creat_error(404, 'File Not Found', e)
+                self.request = Request.Request()
+                self.request.full_request = True
+                self._recv_buff = ""
+            else:
+                self._send_buff = util.creat_error(500, 'Internal Error', e)
+                self.request = Request.Request()
+                self.request.full_request = True
+                self._recv_buff = ""
+        except Exception as e:
+            self.logger.error(traceback.format_exc())
+            raise
+
+    def send(self):
+        """ Fill self.send_buff with all the data it needs and then send it
+        """
+        if self.check_finished_request():
+            raise CustomExceptions.FinishedRequest
+        free_space_in_buffer = self._buff_size - len(self._send_buff)
+        if free_space_in_buffer > 0 and self.file is not None:
+            self._send_buff += self.file.read_buff(free_space_in_buffer)
+        if len(self._send_buff) > 0:
+            maybe_sent_status = False
+            if constants.DOUBLE_CRLF in self._send_buff:
+                maybe_sent_status = True
+            self._send_my_buff()
+            if constants.DOUBLE_CRLF in self._send_buff:
+                maybe_sent_status = False
+            if maybe_sent_status:
+                self.request.sent_status = True
+        if self.check_finished_request():
+            raise CustomExceptions.FinishedRequest
```

Comment:
One of the things I must know is if I already sent the the status.
If I didn't sent status already  I check if there is a status (whice end with \r\n\r\n) loaded in the send_buff, whice is the buff who contain all the data of the client that I need to send.
After this I sent everything I can.
After I finish sending, I check if there a status already (or part of in) loaded in the send_buff and if It was loaded before and now not it means I sent it, and I change that part in the request that  contain the boolean whice mean "a boolean who say if we already sent status or not"